docs/explanation/charm-architecture.md

```diff
@@ -0,0 +1,115 @@
+ }
+    System_Ext(osbuilding, "OpenStack", "OpenStack deployment used for building images")
+
+Container_Boundary(c2, "GitHub Runner"){
+
+    Container(githubrunner, "GitHub Runner Charm", "", "Manages self-hosted runners")
+}
+    Rel(imagebuilder, osbuilding, "builds images")
+    UpdateRelStyle(imagebuilder, osbuilding, $offsetY="-30", $offsetX="10")
+    Rel(imagebuilder, osgithubrunner, "uploads images")
+    UpdateRelStyle(imagebuilder, osgithubrunner, $offsetY="-30", $offsetX="-90")
+
+
+
+
+    Rel(imagebuilder, githubrunner, "image ids")
+    UpdateRelStyle(imagebuilder, githubrunner, $offsetY="-10", $offsetX="-30")
+
+    System_Ext(osgithubrunner, "OpenStack", "OpenStack deployment used for spawning runner VMs")
+    Rel(githubrunner, osgithubrunner, "spawns VMs")
+    UpdateRelStyle(githubrunner, osgithubrunner, $offsetY="-30", $offsetX="10")
+
+    Rel(githubrunner, imagebuilder, "OpenStack credentials")
+    UpdateRelStyle(githubrunner, imagebuilder, $offsetY="10", $offsetX="-60")
+```
+
+
+The image-builder uses the [Openstack SDK](https://docs.openstack.org/openstacksdk/latest/)  to spawn a VM instance in a cloud specified
+by a config option. Using an external OpenStack VM instead of the charms machine allows for more features
+(using chroot has some limitations, e.g. for building snaps) and parallel image building.
+[cloud-init](https://cloud-init.io/) is used to install the necessary dependencies for spawning self-hosted runners
+([github actions runner binary](https://github.com/actions/runner)) and e.g. tools for automatic proxy support ([aproxy](https://github.com/canonical/aproxy)). 
+In addition, software that is heavily used in Canonical projects (such as [juju](https://juju.is/) or [microk8s](https://microk8s.io/)) 
+is pre-installed to reduce the actual CI runtime (as users do not have to install this software in their respective CI runs). 
+There is also a custom script configuration combined with a secret that is run in the cloud-init script to allow further customization of the images.
+The image-builder repeatedly checks to see if the cloud-init script has finished running, then snapshots the VM, uploads the image to a specified Openstack project,
+and deletes the VM. This specified Openstack project is determined via the `image:github_runner_image_v0` integration with another charm (e.g. [GitHub Runner Charm](https://charmhub.io/github-runner)).
+The other charm can then use the image to spawn a VM instance with the necessary software preinstalled.
+The charm sets up a cron job to build the images periodically to ensure that the latest software is installed in the images
+
+
+The interactions between the charm and the image-builder are performed using CLI commands. 
+The image-builder application is not daemonized and stops running after the image has been built
+and uploaded to Openstack.
+
+The image-builder application must be initialized before it can be used. Initialization includes
```

Comment:
It sounds like the user must do this, would it be better to say
```suggestion
The image-builder application is initialized by the charm before it can be used. Initialization includes
```